workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/LayeredDialogSessions.kt

```diff
@@ -113,7 +113,34 @@ public class LayeredDialogSessions private constructor(
 
   private var sessions: List<DialogSession> = emptyList()
 
-  public var allowEvents: Boolean = true
+  /**
+   * Reflects whether [onAttachedToWindow] has been called yet, so that
+   * we can avoid creating `Dialog` windows before the `Activity` window
+   * is ready.
+   */
+  private var attached = false
+
+  /**
+   * Set by an outermost instance only. If we are in [update] before [attached]
+   * is true, holds on to the set of all [DialogSession] to be shown at
+   * attach time, instead of immediately. Necessary to ensure that view state
+   * is restored in the correct order -- the Activity contentView must be attached
+   * to its window before those of any Dialog instance.
+   *
+   * Note that this is a superset of [sessions], includes built by
+   * any nested [LayeredDialogSessions] instances during recursive [update]
+   * calls.
+   *
+   * https://github.com/square/workflow-kotlin/issues/1001
```

Comment:
```suggestion
   * Set by an outermost instance only. If we are in a root [update] call before 
   * [attached] is true, holds on to the set of all [DialogSession] to be 
   * [shown][showNewDialog] at attach time, instead of immediately. Necessary to
   * ensure that view state is restored in the correct order -- the `Activity`
   * `contentView` must be attached to its window before that of any `Dialog`.
   *
   * Note that this is a superset of [sessions], includes built by
   * any nested [LayeredDialogSessions] instances during recursive [update]
   * calls.
   *
   * https://github.com/square/workflow-kotlin/issues/1001
```